hca/full_download.py

```
@@ -0,0 +1,118 @@
+            default="aws"
+        )
+
+    @classmethod
+    def _download_files(cls, files, folder, api, replica):
+        """Use the API class to make a put-files request on each of these files."""
+        for file in files:
+
+            file_uuid = file[cls.UUID]
+            filename = file.get("name", file_uuid)
+
+            eprint("File {}: Retrieving...".format(filename))
+
+            request = [
+                "get-files",
+                file_uuid,
+                "--replica", replica,
+            ]
+
+            response = api.make_request(request, stream=True)
+
+            if response.ok:
+                eprint("File {}: GET successful. Writing to disk.".format(filename, file_uuid))
+                with open(folder + filename, "wb") as fh:
+                    fh.write(response.content)
+                response.close()
+                eprint("File {}: write successful. Stored at {}.".format(filename, folder + filename))
+
+            else:
+                eprint("File {}: GET FAILED. This uuid is neither a bundle nor a file.".format(filename))
+                eprint(response.text)
+                response.close()
+                response.raise_for_status()
+
+        return {"completed": True}
+
+    @classmethod
+    def _download_bundle(cls, args, api):
+        """Use the API class to make a get-bundles request."""
+        bundle_uuid = args["uuid"]
+        bundle_name = args.get("name", bundle_uuid)
+        replica = args["replica"]
+
+        eprint("Bundle {}: Retrieving...".format(bundle_uuid))
+        request = [
+            "get-bundles",
+            bundle_uuid,
+            "--replica", replica,
+        ]
+
+        files = [args]
+        folder = ""
+        response = api.make_request(request)
+
+        if response.ok:
+            files = response.json()["bundle"]["files"]
+            folder = bundle_name + "/"
+            if not os.path.isdir(folder):
+                os.mkdir(folder)
```

Don't see much difference between the two. `mkdir` makes all intermediate folders required, and they both raise an error if the folder already exists. Do you think it should raise an error if the folder already exists?